docs/checkit.js

```diff
@@ -777,55 +786,118 @@ CheckIt.prototype.validateInput = function ($form, $inputFormGroup, inputId) {
+    calendar = new Calendar(calState);
+
+    // Initialize calendar in the storage
+    initP = this.store.initializeCalendar(calendar);
+
+    //add calendar to dropdown
+    this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
+        this.$calendarDropdown);
 
-        //add calendar to dropdown
-        this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
-            this.$calendarDropdown);
+    //build the calendar
+    buildP = this.buildCalendar(calendar);
+    this.hideForm(this.$buildCalendarForm);
     
-        //build the calendar
-        buildP = this.buildCalendar(calendar);
-        this.hideForm(this.$buildCalendarForm);
+    return Promise.all([initP, buildP]);
+    //}
+};
+
+CheckIt.prototype.editCalendar = function (title, start, end, activeCalId) {
+    // Edits the calendar currently on display (current active calendar)
+    // and displays the edited version
+    "use strict";
+    
+    var checkit = this,
+        startDate = moment(start, "YYYY-MM-DD"),
+        endDate = moment(end, "YYYY-MM-DD"),
+        loadP;
+    
+    // I want to return a non-pending promise
+    
+    return checkit.store.loadById(activeCalId)
+        .then(function (calState) {
+            calState.endDateString = endDate.format("YYYYMMDD");
+            calState.startDateString = startDate.format("YYYYMMDD");
+            calState.title = title;
+            checkit.createCalendar(calState);
+        });
+};
+
+
+CheckIt.prototype.editOrCreate = function (params) {
+    // Determine whether to edit an existing calendar
+    // or create a new one
+    
+    // Run edit calendar based on whether a user clicks on a button
+    // with the word "Edit" on it or a button with the word "Create"
+    // on it
+    "use strict";
+
+    var checkit = this,
+        state,
+        start = params.start || this.$startDate.val(),
+        end = params.end || this.$endDate.val(),
+        title = params.title || this.$calendarTitle.val(),
+        buttonId = params.buttonId || event.target.id;
+    
+    if (checkit.validateForm(start, end)) {
         
-        return Promise.all([initP, buildP]);
+        // Check if there is a current active calendar
+        // Don't have to return promise here? TODO
+        if (buttonId === "editButton") {
+            return checkit.store.getActive()
+                .then(function (activeCalId) {
+                    if (activeCalId) { // I don't think I need this condition
+                        // we are editing an existing calendar
+
+                        return checkit.editCalendar(title, start, end, activeCalId);
+                    } else {
+                        console.log("No calendar currently on display");
+                        return Promise.reject("No calendar currently on display");
+                    }
+                })
+                .catch(function (err) { // runs if either getActive fails
```

Comment:
I don't think you need this catch if all it's going to do is return another rejected promise. It's already a rejected promise if you're inside the catch.

Reply:
👍